src/main/java/org/hisp/dhis/integration/icd1x/routes/ICDRouteBuilder.java

```diff
@@ -0,0 +1,143 @@
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.integration.icd1x.routes;
+
+import static org.hisp.dhis.integration.icd1x.Constants.PROPERTY_AUTH_REQUESTED;
+import static org.hisp.dhis.integration.icd1x.Constants.getAsExchangeProperty;
+
+import java.util.Collections;
+import java.util.LinkedList;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.model.dataformat.JsonLibrary;
+import org.hisp.dhis.integration.icd1x.Constants;
+import org.hisp.dhis.integration.icd1x.config.ICDCommandConfig;
+import org.hisp.dhis.integration.icd1x.models.Entity;
+import org.hisp.dhis.integration.icd1x.processors.EnqueueEntitiesProcessor;
+import org.hisp.dhis.integration.icd1x.processors.HeadersSetter;
+import org.hisp.dhis.integration.icd1x.processors.ToOptionsProcessor;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+@Component
+public class ICDRouteBuilder extends RouteBuilder
+{
+    private Processor initRoute()
+    {
+        return exchange -> {
+            ICDCommandConfig routeConfig = exchange.getProperty( Constants.PROPERTY_COMMAND_CONFIG,
+                ICDCommandConfig.class );
+
+            // extracting properties
+            exchange.setProperty( Constants.PROPERTY_HOST, routeConfig.getHost() );
+            exchange.setProperty( Constants.PROPERTY_RELEASE, routeConfig.getReleaseId() );
+            exchange.setProperty( Constants.PROPERTY_LINEARIZATION, routeConfig.getLinearizationName() );
+            exchange.setProperty( Constants.PROPERTY_LANGUAGE, routeConfig.getLanguage() );
+            exchange.setProperty( Constants.PROPERTY_OUTPUT_FILE, routeConfig.getFileOut() );
+            exchange.setProperty( Constants.VERBOSE, routeConfig.isVerbose() );
+            exchange.setProperty( Constants.PROPERTY_ICD_VERSION, routeConfig.getIcdVersion() );
```

Comment:
We are taking them as command-line arguments since this script is mostly a one-time operation. I'm not sure whether command-line arguments from spring shell can be mapped into injectable spring properties. Will check whether it's possible.

Reply:
@claudemamo Any thoughts on this? I assume things are different when it's a service vs a command-line tool.